pageserver/src/layered_repository/disk_btree.rs

```diff
@@ -442,17 +462,22 @@ where
         }
     }
 
-    pub fn append(&mut self, key: &[u8; L], value: u64) -> Result<(), anyhow::Error> {
-        assert!(value <= MAX_VALUE);
+    pub fn append(&mut self, key: &[u8; L], value: u64) -> Result<()> {
+        if value > MAX_VALUE {
+            return Err(DiskBtreeError::AppendOverflow(value));
+        }
         if let Some(last_key) = &self.last_key {
-            assert!(key > last_key, "unsorted input");
+            // FIXME is it ok to have duplicates? should it be just "<"?
```

Comment:
@hlinnaka WDYT? Can this be a cause of https://github.com/neondatabase/neon/issues/1606?

Reply:
Hmm, I don't see how we would get duplicate keys while compacting. Let's print out the keys in the error message, so that we get more info the next time it happens.